src/components/ThemeModal/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+
+// Components
+import { BackButton, HeaderTitle, ModalHeader } from '../Menu/MenuModalWrapper';
+
+// Hooks
+import { useEtherspot } from '../../hooks';
+
+// Icons
+import { IoMdCheckmark } from 'react-icons/io';
+import { CloseButton } from '../Button';
+
+// Utils
+import { THEME_TYPES, Theme, getTheme } from '../../utils/theme';
+
+const ThemeModal = ({ onBackButtonClick }: { onBackButtonClick: () => void }) => {
+  const [appliedTheme, setAppliedTheme] = useState<string>('');
+  const theme: Theme = useTheme();
+  const { changeTheme } = useEtherspot();
+  const themeOptions = Object.keys(THEME_TYPES);
+
+  useEffect(() => {
+    const activeTheme = localStorage.getItem('current-theme');
+    if (activeTheme) {
+      setAppliedTheme(activeTheme);
+      return;
+    }
+    setAppliedTheme(THEME_TYPES.DARK);
+  }, []);
+
+  const onThemeSelection = (themeValue: string) => {
+    setAppliedTheme(themeValue);
+    localStorage.setItem('current-theme', themeValue);
+    changeTheme(getTheme(themeValue));
+  };
+
+  return (
+    <SubMenuWrapper>
+      <ModalHeader style={{ marginBottom: '8px' }}>
+        {onBackButtonClick && (
+          <BackButton color={theme?.color?.background?.settingMenuMain} onClick={onBackButtonClick} />
+        )}
+        <HeaderTitle>Theme</HeaderTitle>
+        <CloseButtonWrapper onClick={onBackButtonClick} />
+      </ModalHeader>
+      {themeOptions.map((value, index) => (
```

Comment:
Regarding comment above
```suggestion
      {Object.keys(ThemeType).map((value, index) => (
```